lib/bolt/transport/winrm.rb

```diff
@@ -62,6 +64,11 @@ def self.validate(options)
           raise Bolt::ValidationError, 'ssl option must be a Boolean true or false'
         end
 
+        basic_auth_only_flag = options['basic-auth-only']
+        unless !!basic_auth_only_flag == basic_auth_only_flag
+          raise Bolt::ValidationError, 'basic-auth-only option must be a Boolean true or false'
+        end
+
```

Comment:
Validation that the option is a Boolean value will be unnecessary since the config transport class automatically validates against a type set in the `OPTIONS` hash.